### PR TITLE
bookstore_cms: add reducers and actions

### PR DESCRIPTION
**Pull request by jmoseka**
I have configured Redux Store and written actions and reducer for adding and removing books in this PR. The following requirements are implemented 

Structured the application files using the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for the Redux files. 
    ```    
    ...
    /src
        |---/components
        |---/redux
            |--- /books
                    | books.js
            |--- /categories
                    | categories.js
            | configureStore.js
        | index.js
    ``` 

- Wrote book's actions and reducer (in `/src/redux/books/books.js`) using the [*ducks* pattern](https://github.com/erikras/ducks-modular-redux). 
    - Define *action types* for adding and removing a book.
    - Set the initial state to be an empty array of books.
    - Export *Action Creators* for your *actions*.
    - Write your reducer and export it as default.
        - Define state changes for the actions that you created.
        - In case of unknown action - return the current state.
        - 
- Wrote categories actions and reducer (in `/src/redux/categories/categories.js`) using the [*ducks* pattern](https://github.com/erikras/ducks-modular-redux). 
    - Define an *action type* for checking the status.
    - Set the initial state to be an empty array of categories.
    - Export *Action Creators* for your *actions*.
    - Write your reducer and export it as default.
        - For the check status action return a string "Under construction".
        - In case of unknown action - return the current state.

- Configure the Redux Store (`/src/redux/configureStore.js`):
    - Importing the necessary methods from Redux.
    - Combine both reducers into a root reducer by using  `combineReducers` function.
    - Use the root reducer to create store.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+          id: Math.floor(Math.random() * 1000),
+          name: action.payload.task,
+        },
+      ];
+    case DELETE_BOOK:
+      return state.filter((item) => item.id !== action.payload.id);
+    default:
+      return initialState;
+  }
+}
```
Comment (ahmetbozaci) [inline src/redux/books/books.js]: Well done with writing reducer, actions with duck patterns.
But ı can't add any book in the page. ı can only see same books from your array.
- [ ] Please fix it. I should add book and delete book from the page.
image

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


 @jmoseka  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Nice to have 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
___


### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Comment by jmoseka** [comment]
Resolved. Add and remove functionalities added

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @jmoseka 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**